OpenTokSamples/1.Hello-World/src/main/java/com/opentok/android/samples/hello_world/MainActivity.java

```diff
@@ -1,289 +0,0 @@
-        mSubscriberViewContainer = (LinearLayout) findViewById(R.id.subscriberview);
-
-        requestPermissions();
-    }
-
-    @Override
-    protected void onStart() {
-        Log.d(TAG, "onStart");
-
-        super.onStart();
-    }
-
-    @Override
-    protected void onRestart() {
-        Log.d(TAG, "onRestart");
-
-        super.onRestart();
-    }
-
-    @Override
-    protected void onResume() {
-        Log.d(TAG, "onResume");
-
-        super.onResume();
-
-        if (mSession == null) {
-            return;
-        }
-        mSession.onResume();
-    }
-
-    @Override
-    protected void onPause() {
-        Log.d(TAG, "onPause");
-
-        super.onPause();
-
-        if (mSession == null) {
-            return;
-        }
-        mSession.onPause();
-
-        if (isFinishing()) {
-            disconnectSession();
-        }
-    }
-
-    @Override
-    protected void onStop() {
-        Log.d(TAG, "onPause");
-
-        super.onStop();
-    }
-
-    @Override
-    protected void onDestroy() {
-        Log.d(TAG, "onDestroy");
-
-        disconnectSession();
-
-        super.onDestroy();
-    }
-
-    @Override
-    public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions, @NonNull int[] grantResults) {
-        super.onRequestPermissionsResult(requestCode, permissions, grantResults);
-
-        EasyPermissions.onRequestPermissionsResult(requestCode, permissions, grantResults, this);
-    }
-
-    @Override
-    public void onPermissionsGranted(int requestCode, List<String> perms) {
-        Log.d(TAG, "onPermissionsGranted:" + requestCode + ":" + perms.size());
-    }
-
-    @Override
-    public void onPermissionsDenied(int requestCode, List<String> perms) {
-        Log.d(TAG, "onPermissionsDenied:" + requestCode + ":" + perms.size());
-
-        if (EasyPermissions.somePermissionPermanentlyDenied(this, perms)) {
-            new AppSettingsDialog.Builder(this, getString(R.string.rationale_ask_again))
-                    .setTitle(getString(R.string.title_settings_dialog))
-                    .setPositiveButton(getString(R.string.setting))
-                    .setNegativeButton(getString(R.string.cancel), null)
-                    .setRequestCode(RC_SETTINGS_SCREEN_PERM)
-                    .build()
-                    .show();
-        }
-    }
-
-    @AfterPermissionGranted(RC_VIDEO_APP_PERM)
-    private void requestPermissions() {
-        String[] perms = { Manifest.permission.INTERNET, Manifest.permission.CAMERA, Manifest.permission.RECORD_AUDIO };
-        if (EasyPermissions.hasPermissions(this, perms)) {
-            mSession = new Session(MainActivity.this, OpenTokConfig.API_KEY, OpenTokConfig.SESSION_ID);
-            mSession.setSessionListener(this);
-            mSession.connect(OpenTokConfig.TOKEN);
-        } else {
-            EasyPermissions.requestPermissions(this, getString(R.string.rationale_video_app), RC_VIDEO_APP_PERM, perms);
-        }
-    }
```

Comment:
@hbaqai  @jeffswartz IMHO we should not remove the current sample but this is up to you guys. At least you guys should not remove functionality. Here you are removing all the logic the sample had to handle permission. I do not see in the new bits being added.